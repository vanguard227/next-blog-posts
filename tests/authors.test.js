import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

import Authors from "../components/authors";

describe("<Authors />", () => {
  const authors = [
    {
      id: "1",
      name: "John Smith",
      avatar: "https://cdn.fakercloud.com/avatars/iamsteffen_128.jpg",
    },
  ];

  it("matches snapshot", () => {
    const component = renderer.create(<Authors authors={authors} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders authors", () => {
    const wrapper = shallow(<Authors authors={authors} />);
    let actual = wrapper.find("img").props().src;
    //expect(actual).toBe(authors[0].avatar);

    actual = wrapper.find("p").text();
    expect(actual).toBe(authors[0].name);

    actual = wrapper.find("div").length;
    expect(actual).toBe(authors.length);
  });
});
