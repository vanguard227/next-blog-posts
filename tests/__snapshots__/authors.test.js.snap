// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authors /> matches snapshot 1`] = `
<div
  className="sc-bczRLJ hWmmZ"
>
  <div>
    <div
      className="sc-gsnTZi ikcVbO"
    >
      <img
        alt="image"
        src="https://i.pinimg.com/564x/64/a5/f2/64a5f243aa5efbea4bb704007459f4c3--create-a-cartoon-annoying-things.jpg"
      />
    </div>
    <p>
      John Smith
    </p>
  </div>
</div>
`;
